pkg/sources/adapter/salesforcesource/auth/jwt.go

```diff
@@ -58,6 +58,11 @@ type claims struct {
 
 // NewJWTAuthenticator creates an OAuth JWT authenticator for Salesforce.
 func NewJWTAuthenticator(certKey, clientID, user, server string, client *http.Client, logger *zap.SugaredLogger) (Authenticator, error) {
+	// JWT package marshals Audience as array even if there is only one element in it. This does not
+	// seem to be supported by Salesforce. By setting the following option to false we tell the imported
+	// library to marshal single item Audience array as a string.
+	jwt.MarshalSingleStringAsArray = false
```

Comment:
This is a **global** setting, so we shouldn't obfuscate it inside a constructor of some sort.

IMO global tweaks should be made obvious inside the `main` or `init`.

Reply:
I have mixed feelings since that global parameter is set for the Salesforce adapter binary only (no other component, nor the controller), and tied to the `github.com/golang-jwt/jwt/v4` library. Moving that to `cmd/salesforcesource-adapter/main` will move the dependency upwards which does not feel great, but will also bring awareness about the parameter.

That reminds me that I need to add this same parameter for the Salesforce target, which I will do in a minute.
Would it work for you adding the param setting to each of the main.go (source and target) but also making main aware of this import?

I don't have a strong opinion on this, this would be problematic if we create a single binary that does JWT and needs `aud` in different styles, not our case, so whatever you feel more intuitive is ok with me.

Reply:
moved it to `main` at both, source and target.
https://github.com/triggermesh/triggermesh/pull/332/commits/b0a7dfd762a15c2a7c610847b33af5c5aa4b87c9

Reply:
If you're very worried about having this obscure line in the `main()`, we can always create a shared helper `salesforce.SetupJWT()` and call it wherever that makes sense.

The only thing which bothered me was the fact that it was obfuscated a level deeper. It's usually considered a good practice to initialize globals _once_ and _early_.

Reply:
agree, I think it is less prone to obscure errors now that it is at main